### DISCUSSION TITLE
Simplify Docker deployment with host IP

### DISCUSSION THREAD

**Discussion by tbendien** [bionic-gpt/bionic-gpt#359] [Ideas]
@ianpurton could you update the docker-compose.yml file to deploy Bionic to have use the local host IP as it's web ui target vs. only localhost? 

I have portable Bionic workstations and I am constantly moving them around and plugging them into different ISP connections on different subnets.

If the Bionic instance could just use the local host IP for the Web and Postgres UI, it would be so much easier to quickly build and deploy Bionic hosts and not have to edit docker-compose.yml files each time.

Ollama Web UI works this way and so is very portable and easy to use....

**Comment by ianpurton** [comment]
I'm not sure how to do this. `docker compose yml` doesn't have variables and can't access. Keycloka needs to know what URL to redirect to.

The enterprise version does allow you to set the host one time.

**Reply by tbendien** [reply]
One method is to change all the entries in the docker-compose.yml to the IP of the local host. That works. It's just a pain when you are building multiple hosts for small deployments. I have a use case where I deploy multiple small GPU workstations in a training room and sometimes have to use different networks so the users can route to the Bionic servers. A lot of times I don't know the IP subnet until I get to the training room, so then have to site there and edit the yaml files...it would be so mush easier if for Community Edition if Bionic just used the local host IP out of the box..like Ollama Web UI. Makes deployments really fast and easy...

**Reply by ianpurton** [reply]
Not sure how ollama web UI handles host ip. Remember ollama web ui is not using keycloak.

I think you could do something like... (from chat-gpt)

```
sed -i 's/localhost/$(ip route get 1 | awk '\''{print $NF; exit}'\'')/g' docker-compose.yml
```

But really you should be using the enterprise version :)

**Reply by votinhthuong** [reply]
Even I'm change all localhost to IP address cannot access it.
